src/main/java/net/pms/configuration/PmsConfiguration.java

```diff
@@ -2720,12 +2722,29 @@ public void setIpFilter(String value) {
 		configuration.setProperty(KEY_IP_FILTER, value);
 	}
 
-	public void setPreventsSleep(boolean value) {
-		configuration.setProperty(KEY_PREVENTS_SLEEP, value);
-	}
-
-	public boolean isPreventsSleep() {
-		return getBoolean(KEY_PREVENTS_SLEEP, true);
+	public void setPreventSleep(PreventSleepMode value) {
+		if (value == null) {
+			throw new NullPointerException("value cannot be null");
+		}
+		configuration.setProperty(KEY_PREVENT_SLEEP, value.getValue());
+		PMS.get().getSleepManager().setMode(value);
+	}
+
+	public PreventSleepMode getPreventSleep() {
+		PreventSleepMode sleepMode = null;
+		String value = getString(KEY_PREVENT_SLEEP, null);
+		if (value == null && configuration.containsKey(KEY_OLD_PREVENTS_SLEEP)) {
+			// Backwards compatibility
+				sleepMode =
```

Comment:
@Nadahar I'm not sure I understand the issue with indentation, I haven't had any issues with that personally. I also don't know how multi-line ternaries can prevent temporary variables in that case but it would be good to see a code example for it.

I've been learning the Go language lately and they actually prescribe formatting and there's a built-in formatting tool in the library, it's a relief to not have to make decisions on this stuff. Oracle used to have a guide but they seem to have stopped updating it. Anyway maybe we can find some code standard to follow and stick to that. Unfortunately the Google style guide doesn't mention multi-line ternaries https://google.github.io/styleguide/javaguide.html

My opinion on the best format is:

```
sleepMode = getBoolean(KEY_OLD_PREVENTS_SLEEP, true) ?
	PreventSleepMode.PLAYBACK :
	PreventSleepMode.NEVER;
```

Once we agree on the format, I think we should record it in a file in the project. Maybe we can say something like:

```
UMS Style Guide:
Things not specified in this guide should use the Google Style Guide at https://google.github.io/styleguide/javaguide.html

Ternaries:
(the style we decide on)
```

We can also consider making a linter config for our styles to make it easier to remember if the list grows.